core/common/src/ContourDisplay.ts

```diff
@@ -0,0 +1,263 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Symbology
+ */
+
+import { compareNumbers, Id64String } from "@itwin/core-bentley";
+import { ColorDef, ColorDefProps } from "./ColorDef";
+import { LinePixels } from "./LinePixels";
+
+/** Namespace containing types controlling how contour lines are rendered.
+ * @public
+ * @extensions
+ */
+export namespace ContourDisplay {
+  /** JSON representation of the rendering styling settings that apply to a specific set of subcategories within a [[ContourDisplay.Terrain]]. */
+  export interface ContourProps {
+    /** See [[Contour.majorColor]]. */
+    majorColor?: ColorDefProps;
+    /** See [[Contour.minorColor]]. */
+    minorColor?: ColorDefProps;
+    /** See [[Contour.majorPixelWidth]]. */
+    majorPixelWidth?: number;
+    /** See [[Contour.minorPixelWidth]]. */
+    minorPixelWidth?: number;
+    /** See [[Contour.majorPattern]]. */
+    majorPattern?: LinePixels;
+    /** See [[Contour.minorPattern]]. */
+    minorPattern?: LinePixels;
+    /** See [[Contour.minorInterval]]. */
+    minorInterval?: number;
+    /** See [[Contour.majorIntervalCount]]. */
+    majorIntervalCount?: number;
+  }
+
+  /** The rendering styling settings that apply to a specific set of subcategories within a [[ContourDisplay.Terrain]].
+   * This actually describes stylings for two sets of contours: major and minor. These stylings are separate from each other.
+   * The minor contour occurs at a defined interval in meters. These intervals draw at a fixed height; they are not dependent on the range of the geometry to which they are applied.
+   * The major contour is dependent on the minor contour. The interval of its occurence is not measured directly in meters; rather it is a count of minor contour intervals between its occurrences.
+   */
+  export class Contour {
+    /** Color that a major contour line will use. Defaults to [[ColorDef.black]].*/
+    public readonly majorColor: ColorDef;
```

Comment:
Did you consider having a type `{ color, width, pattern }` and two properties `major` and `minor` each of that type?

Reply:
One problem with that is that there is not a major/minor interval per se, those are 2 different things (count and interval).

Reply:
`ContourStyleProps` and `ContourStyle` groupings for these properties have been written in latest code push.